src/index.js

```diff
@@ -243,10 +295,10 @@ class Bitswap {
    * Fetch a a list of blocks by cid. If the blocks are in the local
    * blockstore they are returned, otherwise the blocks are added to the wantlist and returned once another node sends them to us.
    *
-   * @param {AsyncIterator<CID>} cids
-   * @param {Object} options
-   * @param {AbortSignal} options.abortSignal
-   * @returns {Promise<AsyncIterator<Block>>}
+   * @param {AsyncIterable<CID>|Iterable<CID>} cids
+   * @param {Object} [options]
+   * @param {AbortSignal} [options.signal]
+   * @returns {AsyncIterable<Block>}
```

Comment:
shouldnt this be AsyncGenerator ?

Reply:
I think this is a big question which I boils down to:

1. Are we committed to supporting an AsyncGenerator API or not ?
2. Should change argument types to `AsyncGenerator`s as well ?
    _I do not see much value in making return types `AsyngGenerators` otherwise  as type checker will treat them as`AsyncIterable`s anyway.

Whatever we decide we should probably apply across the code base

Reply:
Oh there is more things to consider, type here is the TS interface for `AsyncGenerator`

```ts
interface AsyncGenerator<T = unknown, TReturn = any, TNext = unknown> extends AsyncIterator<T, TReturn, TNext> {
    // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn>>;
    return(value: TReturn | PromiseLike<TReturn>): Promise<IteratorResult<T, TReturn>>;
    throw(e: any): Promise<IteratorResult<T, TReturn>>;
    [Symbol.asyncIterator](): AsyncGenerator<T, TReturn, TNext>;
}

type IteratorResult<T, TReturn = any> = IteratorYieldResult<T> | IteratorReturnResult<TReturn>;

interface IteratorYieldResult<TYield> {
    done?: false;
    value: TYield;
}

interface IteratorReturnResult<TReturn> {
    done: true;
    value: TReturn;
}

```

which raises following questions:

1. What should be the type for `TReturn`, specifically it the output position ? Probably `void`.

1. What should be a type for `TNext` & should we even accept arguments to `.next()` ?

1. And then there is the is still this loose `.throw`. What's the contract there ?

So if anything we should probably have a type alias that encodes some of the answers here e.g. `type Stream<T> = AsyncGenerator<T, void, void>`.